### DISCUSSION TITLE
GITHUB_SHA on Windows OS

### DISCUSSION THREAD

**Discussion by tc4rdpp** [community/community#25188] [Actions]
<p>Hello all, </p>
<p>I tried searching for a similar problem in the forum and internet but I could not find anything related.</p>
<p>So, I want get the GITHUB_SHA to use in an npm increase version command</p>
<p>In the Ubuntu OS it works fine:</p>
<pre><code>[...]

runs-on: [ubuntu-latest]
strategy:
matrix:
node-version: [12.x]
steps:
- uses: actions/checkout@v2
- name: Use Node.js ${{ matrix.node-version }}
uses: actions/setup-node@v1
with:
node-version: ${{ matrix.node-version }}
registry-url: https://npm.pkg.github.com/

[...]

- run: npm version prerelease --preid=pre-${GITHUB_SHA::7}

[...]
</code></pre>
<p>However, in the Windows OS, which I also need to use, I get nothing:</p>
<pre><code>runs-on: [windows-latest]
strategy:
matrix:
node-version: [12.x]
steps:
- run: git config --global core.longpaths true
- uses: actions/checkout@v2
- name: Use Node.js ${{ matrix.node-version }}
uses: actions/setup-node@v1
with:
node-version: ${{ matrix.node-version }}
registry-url: https://npm.pkg.github.com/

- run: npm version prerelease --preid=pre-${GITHUB_SHA::7}
</code></pre>
<p>By running cmd /c “set” I can see that the variable has a value set.</p>
<p>Any idea why this happens?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/tc4rdpp">@tc4rdpp</a> When a job runs on windows runner, the default shell for a run step is powershell. <strong>${GITHUB_SHA::7} </strong> could not be recognized in powershell. </p>
<p><img src="https://user-images.githubusercontent.com/24806478/181089310-a544fd68-f2d5-4e37-a2cb-52f809c72af5.png" alt="shell_ps.png" data-base62-sha1="2zJVFeP7R75opg4v4H0vbJYoK1Z" width="618" height="102"></p>
<p>You could specify shell in the run step by adding <strong>shell: bash</strong>. </p>
<pre><code>- run: npm version prerelease --preid=pre-${GITHUB_SHA::7}
shell: bash</code></pre>

**Comment by tc4rdpp** [comment]
<p> It worked flawlessly <a class="mention" href="/yanjingzhu">@yanjingzhu</a> </p>
<p>Thank you for your help.</p>
